base/asyncevent.jl

```diff
@@ -87,21 +87,17 @@ mutable struct Timer
         interval ≥ 0 || throw(ArgumentError("timer cannot have negative repeat interval of $interval seconds"))
 
         this = new(Libc.malloc(_sizeof_uv_timer), ThreadSynchronizer(), true)
-        err = ccall(:uv_timer_init, Cint, (Ptr{Cvoid}, Ptr{Cvoid}), eventloop(), this)
+        err = ccall(:jl_uv_update_timer_start, Cint,
+              (Ptr{Cvoid}, Ptr{Cvoid}, Ptr{Cvoid}, Ptr{Cvoid}, UInt64, UInt64),
```

Comment:
The second type should be `Any`.

Reply:
What's happening is the `unsafe_convert` method below causes `this.handle` to be passed instead of `this` when the argument type is `Ptr{Cvoid}`.

Reply:
```suggestion
              (Ptr{Cvoid}, Any, Ptr{Cvoid}, Ptr{Cvoid}, UInt64, UInt64),
```

Reply:
I made the change, although it hung on "Generating precompile statements..." for me (no CPU activity):

```
Generating precompile statements...
signal (15): Terminated
in expression starting at /opt/julia-dev/contrib/generate_precompile.jl:8
epoll_pwait at /lib64/libc.so.6 (unknown line)
uv__io_poll at /workspace/srcdir/libuv/src/unix/linux-core.c:261
uv_run at /workspace/srcdir/libuv/src/unix/core.c:361
jl_task_get_next at /opt/julia-dev/src/partr.c:421
poptaskref at ./task.jl:568
wait at ./task.jl:575
wait at ./condition.jl:104
wait_readbyte at ./stream.jl:1177
readuntil at ./stream.jl:867 [inlined]
readuntil_string at ./io.jl:645 [inlined]
#readuntil#295 at ./io.jl:768
readuntil at ./io.jl:765 [inlined]
#2 at /opt/julia-dev/contrib/generate_precompile.jl:127
```

Reply:
Strange. Works for me.

Reply:
Huh, yeah, seems like it was just bad state. A rebuild worked for me.